spec/controllers/users/sessions_controller_spec.rb

```diff
@@ -1,4 +1,32 @@
 require "rails_helper"
 
 RSpec.describe Users::SessionsController do
+  before do
+    request.env["devise.mapping"] = Devise.mappings[:user]
+  end
+
+  describe "#new" do
+    context "when the locale is not english and an invalid login alert is added by warden and the devise failure app" do
```

Comment:
Two problems with the sign in page:
1. It would only display the invalid login error message as a flash message.
2. It would not respect the locale for the invalid login error message.

This test ensures that we correct both of those issues.

Reply:
For the locale, I'd prefer to see us figure out how to layer our locale data through so that Warden/Devise can find it, rather than render the template twice. It doesn't have to be a blocking request on this PR.

Good test either way.

Reply:
Rendering the template twice? I don't see where that is happening.

Reply:
I was wrong, ignore this; thought we were re-executing the controller